### PR TITLE
[Linux] Add support for file-picker

### PR DESCRIPTION
**Pull request by joone**
This is part of the file-picker patch, which belongs to Chromium content shell.

Bug=XWALK-950

### CODE REVIEW

File: content/browser/webui/content_web_ui_controller_factory.cc [hunk 1]
```diff
@@ -27,6 +30,9 @@ WebUI::TypeID ContentWebUIControllerFactory::GetWebUIType(
 #if !defined(OS_ANDROID)
       url.host() == kChromeUITracingHost ||
 #endif
+#if defined(USE_AURA) && defined(OS_LINUX)
+	  url.host() == kChromeUIFilePickerHost ||
```
Comment (pozdnyakov) [inline content/browser/webui/content_web_ui_controller_factory.cc]: ident

File: content/browser/webui/content_web_ui_controller_factory.cc [hunk 2]
```diff
@@ -14,6 +14,9 @@
 #include "content/public/browser/web_contents.h"
 #include "content/public/browser/web_ui.h"
 #include "content/public/common/url_constants.h"
+#if defined(USE_AURA) && defined(OS_LINUX)
+#include "xwalk/runtime/browser/ui/webui/file_picker/file_picker_ui.h"
```
Comment (pozdnyakov) [inline content/browser/webui/content_web_ui_controller_factory.cc]: AFAIK  we never include smth from xwalk dir.

Comment (joone) [inline content/browser/webui/content_web_ui_controller_factory.cc]: @pozdnyakov 
There are other case that includes files from xwalk dirs:
```
joone@joone-foss:~/ssd/crosswalk/src/content$ grep -rn xwalk *
browser/web_contents/web_contents_impl.cc:120:#include "xwalk/tizen/browser/media/media_webcontents_observer.h"
browser/webui/content_web_ui_controller_factory.cc:18:#include "xwalk/runtime/browser/ui/webui/file_picker/file_picker_ui.h"
common/content_message_generator.h:76:#include "xwalk/tizen/common/media/media_player_messages.h"
content_browser.gypi:1905:        '<(DEPTH)/xwalk/tizen/browser/media/browser_mediaplayer_manager.cc',
content_browser.gypi:1906:        '<(DEPTH)/xwalk/tizen/browser/media/browser_mediaplayer_manager.h',
content_browser.gypi:1907:        '<(DEPTH)/xwalk/tizen/browser/media/media_webcontents_observer.cc',
content_browser.gypi:1908:        '<(DEPTH)/xwalk/tizen/browser/media/media_webcontents_observer.h',
content_browser.gypi:1909:        '<(DEPTH)/xwalk/tizen/browser/media/murphy_mainloop.cc',
content_browser.gypi:1910:        '<(DEPTH)/xwalk/tizen/browser/media/murphy_mainloop.h',
content_browser.gypi:1911:        '<(DEPTH)/xwalk/tizen/browser/media/murphy_resource.cc',
content_browser.gypi:1912:        '<(DEPTH)/xwalk/tizen/browser/media/murphy_resource.h',
content_browser.gypi:1913:        '<(DEPTH)/xwalk/tizen/browser/media/murphy_resource_manager.cc',
content_browser.gypi:1914:        '<(DEPTH)/xwalk/tizen/browser/media/murphy_resource_manager.h',
content_common.gypi:994:        '<(DEPTH)/xwalk/tizen/common/media/media_player_messages.h',
content_renderer.gypi:855:        '<(DEPTH)/xwalk/tizen/renderer/media/mediaplayer_impl.cc',
content_renderer.gypi:856:        '<(DEPTH)/xwalk/tizen/renderer/media/mediaplayer_impl.h',
content_renderer.gypi:857:        '<(DEPTH)/xwalk/tizen/renderer/media/renderer_mediaplayer_manager.cc',
content_renderer.gypi:858:        '<(DEPTH)/xwalk/tizen/renderer/media/renderer_mediaplayer_manager.h',
renderer/render_frame_impl.cc:175:#include "xwalk/tizen/renderer/media/renderer_mediaplayer_manager.h"
renderer/render_frame_impl.cc:176:#include "xwalk/tizen/renderer/media/mediaplayer_impl.h"
```
If it is not allowed, I have to add file-picker to under content or add a webui factory to xwalk

**Comment by PeterWangIntel** [comment]
We can avoid to modify content through "content::WebUIControllerFactory::RegisterFactory" (refer to 'chrome_browser_main.cc')

**Comment by joone** [comment]
@PeterWangIntel  I may be able to make a WebUIControllerFactory for Crosswalk if we have more WebUIs.

**Pull request closed without merging**